src/pages/en/reference/directives-reference.md

```diff
@@ -67,6 +67,25 @@ const cmsContent = await fetchHTMLFromMyCMS();
 <Fragment set:html={cmsContent}>
 ```
 
+`set:html={Promise<string>}` injects an HTML string into an element that is wrapped in a promise.
+
+This can be used to inject HTML stored externally, such as a database.
+
+```astro
+---
+import api from '../db/api.js';
+---
+<article set:html={api.getArticle(Astro.props.id)}></article>
+```
+
+`set:html={Promise<Response>}` injects a [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) into an element.
+
+This is most useful for use with `fetch()`. One use-case is fetching old posts from a previous static-site generator.
```

Comment:
Not completely sure, but looks like the correct spelling is "use case" (without a hyphen), though we do have both "use-case" and "use case" in the docs, the latter having more appearances.
```suggestion
This is most useful for use with `fetch()`. One use case is fetching old posts from a previous static-site generator.
```